upload-chunked-file.js

```diff
@@ -0,0 +1,43 @@
+function chunkFileUpload(file, serverUrl, chunkSize) {
```

Comment:
**TypeScript Type Safety**: Although this is JavaScript, if refactoring in TypeScript, avoid using `any` types and leverage TypeScript's typing features for better code reliability. Define interfaces for the file and chunk metadata.